tasks/rh-sign-image/rh-sign-image.yaml

```diff
@@ -0,0 +1,88 @@
+      description: Path to the JSON string of the mapped Snapshot spec in the data workspace
+      type: string
+      default: "mapped_snapshot.json"
+    - name: dataPath
+      description: Path to the JSON string of the merged data to use in the data workspace
+      type: string
+      default: "data.json"
+    - name: requester
+      type: string
+      description: Name of the user that requested the signing, for auditing purposes
+    - name: requestTimeout
+      type: string
+      default: "180"
+      description: InternalRequest timeout
+  workspaces:
+    - name: data
+      description: workspace to read and save files
+  steps:
+    - name: sign-image
+      image: quay.io/hacbs-release/release-utils:a9603c0ae1982be5e1142d44765b1475adc7ad40
+      script: |
+        #!/usr/bin/env sh
+        #
+        SNAPSHOT_PATH=$(workspaces.data.path)/$(params.snapshotPath)
+        RUNDIR="$(workspaces.data.path)/$(context.taskRun.uid)"
+
+        if [ ! -d $RUNDIR ]; then
+          mkdir -p $RUNDIR
+        fi
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+
+        default_pipeline_image="quay.io/redhat-isv/operator-pipelines-images:released"
+        request=$(jq -r '.sign.request // "hacbs-signing-pipeline"' ${DATA_FILE})
+        pipeline_image=$(jq -r --args default_pipeline_image "${default_pipeline_image}" \
+            '.sign.pipelineImage // "$default_pipeline_image"' ${DATA_FILE})
+        config_map_name=$(jq -r '.sign.configMapName // "signing-config-map"' ${DATA_FILE})
+        synchronously=$(jq -r '.sign.synchronously // "false"' ${DATA_FILE})
+
+        COMPONENTS_LENGTH=$(jq '.components |length' ${SNAPSHOT_PATH})
+        for (( COMPONENTS_INDEX=0; COMPONENTS_INDEX<COMPONENTS_LENGTH; COMPONENTS_INDEX++ )); do
+
+            reference=$(jq -r ".components[${COMPONENTS_INDEX}].containerImage" ${SNAPSHOT_PATH})
+            manifest_digest="${reference#*@}"
+
+            echo -en "Creating InternalRequest to sign image... "
+            internal-request -r "${request}" \
+                -p pipeline_image=${pipeline_image} \
+                -p reference=${reference} \
+                -p manifest_digest=${manifest_digest} \
+                -p requester=$(params.requester) \
+                -p config_map_name=${config_map_name} \
+                -t $(params.requestTimeout) -s ${synchronously} \
+                > ${RUNDIR}/ir-result.txt || \
+                (grep "^\[" ${RUNDIR}/ir-result.txt | jq . && exit 1)
+
+            internalRequest=$(awk 'NR==1{ print $2 }' ${RUNDIR}/ir-result.txt | xargs)
+            echo "done (${internalRequest})"
+
+            status=$(kubectl get internalrequest $internalRequest -o=jsonpath='{.status}')
+            reason=$(jq ".reason" <<< "${status}" )
+            if [[ "${reason}" =~ Failed|Rejected ]]; then
```

Comment:
I said this elsewhere. If the internal request fails, this script will already exit on line 74, no? And in that case we might not even need any of that `awk` and the `kubectl get`.

Reply:
It wont as you teach me with the `||`.

Reply:
If the `internal-request` script fails, the part after `||` will be executed: `(grep "^\[" ${RUNDIR}/ir-result.txt | jq . && exit 1)` which will stop the whole execution of the step (`exit 1`). Or am I missing something?

Reply:
Oh I see. The intention was that in async mode, there will be node status in the output, right? But I actually think it won't work as intended.

It seems the idea is that if the grep is unsuccessful, the `exit` won't be executed. But see this:
```sh
$ grep bla bla|jq . && echo exit 1
grep: bla: No such file or directory
exit 1
```

That's because the result of `grep bla bla|jq .` will always be determined by `jq` which will never fail.
If you want `grep` to fail the whole thing, you would need `pipefail`:
```sh
$ set -o pipefail
$ grep bla bla|jq . && echo exit 1
grep: bla: No such file or directory
```

Reply:
In the specific case, the `ir-result.txt` will always exist with the error I want to parse. Unless something changes in the `internal-request` script and the expected output is not printed anymore, or even we have some  communication error that the script can not get the error - but this would be a unexpected situation and it would cause an error anyways.

This was heavily tested in `run-files-updates` , btw https://github.com/redhat-appstudio/release-service-catalog/blob/main/tasks/run-file-updates/run-file-updates.yaml

Reply:
> In the specific case, the ir-result.txt will always exist with the error I want to parse. 

But only for sync mode, no? In async mode, there will be no result provided by the `internal-request` script.

Reply:
> > In the specific case, the ir-result.txt will always exist with the error I want to parse.
> 
> But only for sync mode, no? In async mode, there will be no result provided by the `internal-request` script.

For this specific case, it is only intended to allow testing. My initial idea was also implement the async now, but @scoheb said yesterday in the stand up to only proceed with the sync mode.

Reply:
I understand that. And it's fine. But what I mean is that the whole point of this snippet is to allow async mode. And as I'm showing, it's not working correctly. And your answer is that in this case, it's always sync, because async is not supported for now. So if it's not supported, we should remove that code. At the very least there should be a comment somewhere saying that async is not supported yet. But really, the script shouldn't even allow it if it's not supported yet.

Reply:
I changed the task to use sync by default and removed for now any possibility to use it async.